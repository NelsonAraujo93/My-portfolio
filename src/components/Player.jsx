import PropTypes from 'prop-types';

function Player({ data }) {
  return (
    <article className="player-card">
      <h3>{data.firstname}</h3>
      <h3>{data.lastname}</h3>
      <h4>{data.leagues.standard.jersey || 'N/A'}</h4>
      <h4>{data.leagues.standard.pos}</h4>
      <h5>{data.birth.date}</h5>
    </article>
  );
}

Player.propTypes = {
  data: PropTypes.instanceOf(Object).isRequired,
};

export default Player;
