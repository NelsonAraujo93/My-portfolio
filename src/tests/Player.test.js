import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Player from '../components/Player';

describe('Tests for <Player /> component', () => {
  const data = {
    firstname: 'Nelson',
    lastname: 'Araujo',
    leagues: { standard: { jersey: 23, pos: 'FC' } },
    birth: { date: '1996-10-24' },
  };

  it('Should match the snapshot', () => {
    expect(renderer.create(<Player data={data} />).toJSON()).toMatchSnapshot();
  });

  it('Should render all rockets children', () => {
    render(<Player data={data} />);
    expect(screen.queryByTestId('player-container').children.length).toBe(5);
  });
});
