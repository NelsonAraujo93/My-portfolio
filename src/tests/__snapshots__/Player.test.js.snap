// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <Player /> component Should match the snapshot 1`] = `
<article
  className="player-card"
  data-testid="player-container"
>
  <h3>
    Nelson
  </h3>
  <h3>
    Araujo
  </h3>
  <h4>
    23
  </h4>
  <h4>
    FC
  </h4>
  <h5>
    1996-10-24
  </h5>
</article>
`;
